authentication_test.go

```diff
@@ -87,17 +87,20 @@ func (s *AuthenticationTestSuite) TestGetAuthURLAllParams() {
 	make := "TESLA"
 	state := "state"
 	VIN := "123456789901234567"
+	Country := "US"
```

Comment:
should we a test where we don't specify country?

Reply:
just seeing all your comments. I think i repeated them, that said, i held back on this one. That is because i dont think we have written tests for optional parameters before this either (I for sure havent in ruby). Not sure if we should change that.] and atleast unit tests.

Reply:
I'm fine either way, I mostly just added this because I don't know go and wanted to be sure that optional params in structs work hahah